chains/evm/contracts/libraries/USDCSourcePoolDataCodec.sol

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+/// @notice Library for encoding and decoding the source pool data for the USDC token pool based on the CCTP version.
+/// @dev While every version of a CCTP-enabled pool has a different source pool data format, the encoding and decoding
+/// schemes are similar in a few ways. Each source pool data format is prefixed with a version number, in bytes4 format.
+/// It is then encodePacked() with each of the sourceTokenDataPayload fields. Decoding means parsing the bytes array
+/// into the corresponding sourceTokenDataPayload struct by parsing each field individually and then assembling them
+/// into the corresponding struct. This adds some additional gas overhead during decoding, but the benefits of saving
+/// space on the source pool data outweigh the overhead.
+/// @dev Any future versions of CCTP should include in this library a new function for encoding and decoding the source
+/// pool data accordingly.
+library USDCSourcePoolDataCodec {
+  /// @dev The flag used to indicate that the source pool data is coming from a chain that does not have CCTP Support,
+  /// and so the lock release pool should be used. The BurnMintWithLockReleaseTokenPool uses this flag as its source pool
+  /// data to indicate that the tokens should be released from the lock release pool rather than attempting to be minted
+  /// through CCTP.
+  /// @dev The preimage is bytes4(keccak256("NO_CCTP_USE_LOCK_RELEASE")).
+  bytes4 public constant LOCK_RELEASE_FLAG = 0xfa7c07de;
+
+  /// @dev The preimage is bytes4(keccak256("CCTP_V1"))
+  bytes4 public constant CCTP_VERSION_1_TAG = 0xf3567d18;
+
+  /// @dev There are two different tags for CCTP V2 to allow for CCIP V1.7 Compatibility which will enable fast transfers.
+  /// Both tags will route to the same CCTP V2 pool, but will allow for pools to identify the type of transfer (slow or fast).
+
+  /// @dev The preimage is bytes4(keccak256("CCTP_V2"))
+  bytes4 public constant CCTP_VERSION_2_TAG = 0xb148ea5f;
+
+  /// @dev The preimage is bytes4(keccak256("CCTP_V2_FAST_TRANSFER"))
+  bytes4 public constant CCTP_VERSION_2_FAST_TRANSFER_TAG = 0x0458016c;
```

Comment:
The new tag is not for fast transfers, it's to signal CCV usage. Both fast and slow can use CCVs and both fast and slow should end up in the same pool

Reply:
https://github.com/smartcontractkit/chainlink-ccip/pull/1134/commits/4aa8aa0fd730a2c6fa961a4f1bc5ad35a65fb086#diff-510793de9f7a7a1b0dd14ec9abdc694fe11b9eb710c18b19be666c59bc459840R31